src/app/wallets/update-pending-payments.ts

```diff
@@ -0,0 +1,187 @@
+import { PaymentStatus } from "@domain/bitcoin/lightning"
+import { InconsistentDataError } from "@domain/errors"
+import { toLiabilitiesAccountId } from "@domain/ledger"
+import { FeeReimbursement } from "@domain/ledger/fee-reimbursement"
+import { LedgerService } from "@services/ledger"
+import { LndService } from "@services/lnd"
+import { LockService } from "@services/lock"
+import { PriceService } from "@services/price"
+
+export const updatePendingPayments = async ({
+  walletId,
+  logger,
+  lock,
+}: {
+  walletId: WalletId
+  logger: Logger
+  lock?: DistributedLock
+}): Promise<void | ApplicationError> => {
+  const liabilitiesAccountId = toLiabilitiesAccountId(walletId)
+
+  const lockService = LockService()
+  await lockService.lockWalletId({ walletId, logger, lock }, async () => {
+    const ledgerService = LedgerService()
+    const pendingPaymentTransactions = await ledgerService.listPendingPayments(
+      liabilitiesAccountId,
+    )
+    if (pendingPaymentTransactions instanceof Error) return pendingPaymentTransactions
+
+    for (const paymentLiabilityTx of pendingPaymentTransactions) {
+      await updatePendingPayment({ liabilitiesAccountId, paymentLiabilityTx, logger })
+    }
+  })
+}
+
+const updatePendingPayment = async ({
+  liabilitiesAccountId,
+  paymentLiabilityTx,
+  logger,
+}: {
+  liabilitiesAccountId: LiabilitiesAccountId
+  paymentLiabilityTx: LedgerTransaction
+  logger: Logger
+}): Promise<void | ApplicationError> => {
+  const paymentLogger = logger.child({
+    topic: "payment",
+    protocol: "lightning",
+    transactionType: "payment",
+    onUs: false,
+    payment: paymentLiabilityTx,
+  })
+
+  const lndService = LndService()
+  if (lndService instanceof Error) return lndService
+
+  const { paymentHash, pubkey } = paymentLiabilityTx
+  // If we had PaymentLedgerType => no need for checking the fields
```

Comment:
We can potentially add things like this as `good first issues` to get open contributors familiar with how our typing works